packages/app/src/pages/SwapPage/PricePerToken.tsx

```diff
@@ -0,0 +1,66 @@
+import { toNumber } from "fuels";
+import { useAtomValue } from "jotai";
+import { useState } from "react";
+import { BiRefresh } from "react-icons/bi";
+
+import { swapIsTypingAtom } from "./jotai";
+
+import { Button } from "~/components/Button";
+
+const style = {
+  wrapper: `flex items-center gap-3 my-4 px-2 text-sm text-gray-400`,
+};
+
+type Direction = "from" | "to";
+
+function getPricePerToken(
+  direction: Direction,
+  fromAmount?: bigint | null,
+  toAmount?: bigint | null
+) {
+  if (!fromAmount || !toAmount) return "";
+  const from = toNumber(fromAmount);
+  const to = toNumber(toAmount);
+  const price = direction === "from" ? from / to : to / from;
+  return price.toFixed(6);
+}
+
+type PricePerTokenProps = {
+  fromCoin?: string;
+  fromAmount?: bigint | null;
+  toCoin?: string;
+  toAmount?: bigint | null;
+};
+
+export function PricePerToken({
+  fromCoin,
+  fromAmount,
+  toCoin,
+  toAmount,
+}: PricePerTokenProps) {
+  const [direction, setDirection] = useState<Direction>("from");
+  const isTyping = useAtomValue(swapIsTypingAtom);
```

Comment:
Why do we need to use `useAtomValue`?

I think you could check directly if the amount is null like on line 53.

Reply:
Because I want to remove the calculation when the user is typing on input. By using the amount directly this glitches calculation when the user is typing. With this state I can check when user is typing and hide this component.